libsrc/Itk2DicomConverter.cpp

```diff
@@ -483,10 +494,117 @@ namespace dcmqi {
         segmentsOverlap = "NO";
       else
         segmentsOverlap = "UNDEFINED";
-      CHECK_COND(segdocDataset.putAndInsertString(DCM_SegmentsOverlap, segmentsOverlap.c_str()));
+      CHECK_COND(segdocDataset->putAndInsertString(DCM_SegmentsOverlap, segmentsOverlap.c_str()));
+    }
+
+    if (sortByLabelID)
+    {
+      sortByLabel(segdocDataset.get(), segNum2Label);
```

Comment:
I do not see how this could possibly work. I do not see any checks for non-consecutive segment numbering, so if there are gaps in label numbers, the result will violate the conventions for having consecutive values for `SegmentNumber`. Did I miss something?

Reply:
I think you are right, this is not addressed.  Probably it can easily by done by either checking the labels in in segNum2Label, or iterating over the segments in Segment Sequence once mapping has been performed.